### PR TITLE
enable lazyloading of tracks when list is long

### PR DESCRIPTION
**Pull request by brentpicasso**
Limit initial loading to X when browsing tracks to improve responsiveness. When scrolling near the bottom, add Y additional tracks on the fly, until the entire list is shown.

### CODE REVIEW

File: autosportlabs/racecapture/views/tracks/tracksview.kv [hunk 1]
```diff
@@ -125,6 +125,7 @@
     	spacing: dp(30)
 		size_hint_y: 0.68
 		ScrollView:
+            on_scroll_move: root.on_scroll(*args)
```
Comment (ryandoherty) [inline autosportlabs/racecapture/views/tracks/tracksview.kv]: This indentation doesn't look quite right.

File: autosportlabs/racecapture/views/tracks/tracksview.py [hunk 2]
```diff
@@ -145,10 +145,36 @@ class TracksBrowser(BoxLayout):
...
+        
+    def on_scroll(self, instance, value):
+        scroll_y = self.ids.scrltracks.scroll_y
+        last_scroll_y = self.last_scroll_y
+        self.last_scroll_y = scroll_y 
+        #only check to lazy load if we're scrolling towards the bottom
+        if  scroll_y < last_scroll_y:
+            self.lazy_load_more_maybe()
+        
+    def lazy_load_more_maybe(self):
```
Comment (ryandoherty) [inline autosportlabs/racecapture/views/tracks/tracksview.py]: This begs for some Carly Rae Jepsen.

**Comment by ryandoherty** [comment]
Getting a crash when I click on the home screen race tracks icon.

**Comment by ryandoherty** [comment]
Also getting a crash when I try to add a track to the RCP configuration or select tracks from the side slider menu.

**Comment by ryandoherty** [comment]
kivy.lang.BuilderException: Parser: File "autosportlabs/racecapture/views/tracks/tracksview.kv", line 128:
 ...
     126:               size_hint_y: 0.68
     127:               ScrollView:
 >>  128:            on_scroll_move: root.on_scroll(*args)
     129:                       id: scrltracks
     130:                   do_scroll_x:False
 ...
 KeyError: 'scroll_move'

**Comment by brentpicasso** [comment]
You're still on Kivy 1.8. needs kivy 1.9 to get the scroll events.

**Comment by ryandoherty** [comment]
Huh, but main.py still has `kivy.require('1.8.0')`. The 1.9.0 update should probably be a separate commit or PR and the app should be tested that everything works. And the Readme updated.

**Comment by brentpicasso** [comment]
True, can add that as a separate PR.

On Tue, Jun 30, 2015 at 1:32 PM, Ryan Doherty <notifications@github.com>
wrote:

> Huh, but main.py still has kivy.require('1.8.0'). The 1.9.0 update should
> probably be a separate commit or PR and the app should be tested that
> everything works. And the Readme updated.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture_App/pull/65#issuecomment-117333476>
> .
>



-- 
Brent Picasso
*Autosport Labs <http://www.autosportlabs.com>*
*Technology for Race and Street*

**Comment by ryandoherty** [comment]
Tested with Kivy 1.9.0, works great!

**Comment by brentpicasso** [comment]
thanks! that was a bit of trickery and changed a couple of times :)

It's a little choppy on android as it loads, but is still usable.

**Pull request merged**